canonical-web-and-design/maas-ui
# Break down MachineList into smaller components.

## Done
- Separated MachineListControls (the filter accordion, searchbox and grouping dropdown) and MachineListTable (the actual table) from MachineList.
- Separated tests accordingly
- Renamed `searchText` to `filter` in MachineListTable to differentiate it from the text that shows up in MachineListControls.
- Added debounce to searchbox (but not filter dropdown)

## QA
- Type in the search box and check that the input is immediate and a spinner shows up.
- Check that after a short period the machine list updates according to the filter.
- Check that there is no spinner when using the filter accordion.
- Nothing else should have changed.

## Fixes
Fixes #1005
